### PR TITLE
ARROW-16489: [R] wrong encoding causes parsing error

### PR DESCRIPTION
**Pull request by assignUser**

### CODE REVIEW

**Comment by assignUser** [comment]
@github-actions crossbow submit test-r-linux-as-cran

**Review by nealrichardson — CHANGES_REQUESTED** [review]
I could be wrong but I don't think this is the right change. This file hasn't changed recently: https://github.com/apache/arrow/commits/master/r/tests/testthat/latin1.R

What did change was how R's parser handles it: https://github.com/wch/r-source/commit/48b9036b120d7ac82c9aa61e6d47225de88d6eb2

https://github.com/apache/arrow/blob/master/r/tests/testthat/test-utf.R#L19-L24 explains how and why this file is set up like this. I would guess the right fix is there. 

Or, you could look at the PR that originally added this (#7527), locally remove the `Rf_translateCharUTF8`s that are added there, and confirm that all of the tests in this file fail with your encoding change here. That way we'd know that we're still covering the quirky behavior that led us to set up the test in this unnatural way.

**Comment by nealrichardson** [comment]
Hmm, this is also failing on CRAN checks now: https://cran.r-project.org/web/checks/check_results_arrow.html

Which means that our `skip_on_cran()` to avoid sourcing this file on CRAN isn't protecting us anymore. So maybe this is a good fix after all, provided that it is still covering the encoding quirks.

**Comment by assignUser** [comment]
Looks like `skip_on_cran` does still work but the error pops while the test are checked for unstated dependencies.  

I removed the `iconv("Veitingastaðir", to = "latin1")` bit and tested it with the string directly and the setup tests detect the wrong encoding, so as far as I understand them the tests should still test the correct behavior.

Should we remove the wrapper and move everything into `test-utf.R`?

**Comment by nealrichardson** [comment]
> Looks like `skip_on_cran` does still work but the error pops while the test are checked for unstated dependencies.
> 

Ah, got it, that makes "sense".

> I removed the `iconv("Veitingastaðir", to = "latin1")` bit and tested it with the string directly and the setup tests detect the wrong encoding, so as far as I understand them the tests should still test the correct behavior.
> 
> Should we remove the wrapper and move everything into `test-utf.R`?

Yeah, if trying to conceal a wonky file from CRAN isn't working anymore, then we should just move it in to test-utf.R, and we probably don't need to skip_on_cran either (though let's verify). 

If you're removing latin1.R, search the repo for the couple of other places where it is referenced and update those too.

**Comment by assignUser** [comment]
@github-actions crossbow sub test-r-linux-as-cran

**Comment by assignUser** [comment]
@github-actions crossbow submit test-r-hub-debian-gcc-devel-lto-latest

**Comment by assignUser** [comment]
@github-actions crossbow submit test-r-rhub-debian-gcc-devel-lto-latest

File: r/tests/testthat/test-utf.R [hunk 1]
```diff
@@ -17,8 +17,63 @@
...
+  expect_identical(as.data.frame(record_batch(df)), df)
+  expect_identical(as.data.frame(record_batch(df_struct)), df_struct)
+
+  # Schema field name
+  df_schema <- do.call(schema, raw_schema)
+  expect_identical(names(df_schema), names(df))
+
+  df_struct_schema <- schema(a = do.call(struct, raw_schema))
+  # StructType doesn't expose names (in C++)
+  # expect_identical(names(df_struct_schema$a), names(df))
```
Comment (nealrichardson) [inline r/tests/testthat/test-utf.R:61]: Looks like you need to resolve this lint check

Comment (nealrichardson) [inline r/tests/testthat/test-utf.R:61]: Looks like you need to resolve this lint check

**Review by nealrichardson — COMMENTED** [review]
LGTM, aside from the lint issue

**Comment by nealrichardson** [comment]
@github-actions crossbow submit test-r-linux-as-cran

**Review by nealrichardson — APPROVED** [review]

File: r/.styler_excludes.R [hunk 2]
```diff
@@ -15,4 +15,4 @@
 #  specific language governing permissions and limitations
 #  under the License.
 
-c("tests/testthat/latin1.R", "data-raw/codegen.R")
\ No newline at end of file
+c("data-raw/codegen.R")
```
Comment (thisisnic) [inline r/.styler_excludes.R:18]: Can we remove the `c()` now there's only one thing here?

Comment (assignUser) [reply r/.styler_excludes.R:18]: Sure, I left it like this to communicate how to add other files (as vector vs line by line similar to .ignore-files) but it's an R file so maybe that is clear enough.

**Review by thisisnic — APPROVED** [review]

**Comment by nealrichardson** [comment]
I still support this change, but three notes for the historical record:

* An R Core developer messaged me about this failure, suggesting that we needed to update our code to handle the classed error that the parser is now returning, and I responded that the code that was causing the issue was in the check for unstated dependencies in tests, i.e. it's not in `arrow`. He responded that there was indeed something in the check code that needed to be updated to handle this and is looking into it. So it is possible that this r-devel check failure will go away on its own soon.
* As this [insightful blog post](https://enpiar.com/2018/02/01/back-to-the-future/#iii-way-back-to-the-distant-past) notes (and as alluded to in a code comment), the reason for the indirection of having a separate latin1.R file was because Solaris choked on non-ASCII files. Since CRAN doesn't check on Solaris anymore, this is no longer a concern, and consolidating the files should be ok.
* When I submitted 8.0.0 to CRAN yesterday, I deleted latin1.R and test-utf.R to avoid the warning.

**Pull request closed without merging**